### PR TITLE
Compatible with the versions after dubbo-2.7.14 

### PR DESCRIPTION
**Pull request by wuwen5**
[The ```rpcContext.getAttachments()``` method returns is  copy Map](https://github.com/apache/skywalking/issues/8415)

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/8415.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/dubbo-2.7.x-plugin/pom.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <dubbo.version>2.7.0</dubbo.version>
+        <dubbo.version>2.7.15</dubbo.version>
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/dubbo-2.7.x-plugin/pom.xml:35]: I don't think this is the correct change. Java agent plugin covers different versions. You change the codes according to the latest version, which makes everyone using < 2.7.14 + SkyWalking agent 8.9.0+ faces unexpected bugs.

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/dubbo-2.7.x-plugin/pom.xml:35]: This kind of API behavior change is strange in this tiny bug-fix version change. Could you report to Dubbo community for an explanation?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/dubbo-2.7.x-plugin/pom.xml:35]: If this API exists for all previous 2.7.x versions. please don't upgrade this dependency version.

Comment (wuwen5) [reply apm-sniffer/apm-sdk-plugin/dubbo-2.7.x-plugin/pom.xml]: Agree

**Comment by wu-sheng** [comment]
Please update the changes.md in the root.

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/dubbo-2.7.x-scenario/support-version.list#L17

The new in-compatible version should be added here to verify this fix.

File: test/plugin/scenarios/dubbo-2.7.x-scenario/support-version.list [hunk 2]
```diff
@@ -14,4 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-2.7.3
\ No newline at end of file
+2.7.3
+2.7.14
+2.7.15
```
Comment (wu-sheng) [inline test/plugin/scenarios/dubbo-2.7.x-scenario/support-version.list:19]: Do 2.7.14 and 2.7.15 have any difference to test?

**Review by wu-sheng — APPROVED** [review]
If all versions of plugin tests pass, we should be good.

File: test/plugin/scenarios/dubbo-2.7.x-scenario/support-version.list (hunk 2 above)
Comment (wuwen5) [reply test/plugin/scenarios/dubbo-2.7.x-scenario/support-version.list]: No, just because it's the latest version, I'll removed it.

**Pull request merged**